pkgs/development/compilers/dotnet/default.nix

```diff
@@ -30,21 +27,57 @@ let
     "i686-windows" = "win-x86";
   };
 
-  # Convert a "stdenv.hostPlatform.system" to a dotnet RID
-  systemToDotnetRid = system: runtimeIdentifierMap.${system} or (throw "unsupported platform ${system}");
-in
-{
-  inherit systemToDotnetRid;
-
   combinePackages = attrs: callPackage (import ./combine-packages.nix attrs) {};
 
-  dotnet_8 = recurseIntoAttrs (callPackage ./8 { bootstrapSdk = dotnet_8_0_102.sdk_8_0; });
-  dotnet_9 = recurseIntoAttrs (callPackage ./9 {});
-} // lib.optionalAttrs config.allowAliases {
-  # EOL
-  sdk_2_1 = throw "Dotnet SDK 2.1 is EOL, please use 6.0 (LTS) or 7.0 (Current)";
-  sdk_2_2 = throw "Dotnet SDK 2.2 is EOL, please use 6.0 (LTS) or 7.0 (Current)";
-  sdk_3_0 = throw "Dotnet SDK 3.0 is EOL, please use 6.0 (LTS) or 7.0 (Current)";
-  sdk_3_1 = throw "Dotnet SDK 3.1 is EOL, please use 6.0 (LTS) or 7.0 (Current)";
-  sdk_5_0 = throw "Dotnet SDK 5.0 is EOL, please use 6.0 (LTS) or 7.0 (Current)";
-} // dotnet_6_0 // dotnet_7_0 // dotnet_8_0 // dotnet_9_0
+  makeDotnet = args:
+    let self =
+      lib.makeOverridable ({ sdk, runtime, aspnetcore, ...}@pkgs:
+        recurseIntoAttrs (
+          pkgs // {
+            buildDotnetModule = buildDotnetModule.override { dotnet = self; };
+            buildDotnetGlobalTool = buildDotnetGlobalTool.override { dotnet = self; };
+            nuget-to-nix = nuget-to-nix.override { dotnet = self; };
+
+            withExtraSDKs = sdks: self.override {
+              sdk = combinePackages (lib.singleton sdk ++ sdks);
+            };
+          })) args;
+    in self;
+
+  pkgs = rec {
+    # Convert a "stdenv.hostPlatform.system" to a dotnet RID
+    systemToDotnetRid = system: runtimeIdentifierMap.${system} or (throw "unsupported platform ${system}");
+
+    inherit combinePackages;
+
+    ## Files in versions/ are generated automatically by update.sh ##
+    dotnet_6-bin = makeDotnet (import ./versions/6.0.nix buildAttrs);
+    dotnet_7-bin = makeDotnet (import ./versions/7.0.nix buildAttrs);
+    dotnet_8-bin = makeDotnet (import ./versions/8.0.nix buildAttrs);
+    dotnet_8_0_102-bin = makeDotnet (import ./versions/8.0.102.nix buildAttrs);
```

Comment:
maybe name this `dotnet_8_bootstrap` or something similar, so the purpose of the package is obvious (and it doesn't get used when not needed by accident). Or at least add a comment.

Reply:
I don't see why they would be different versions at all. .NET 8 SDK 8.0.201 and 8.0.102 are both part of the .NET 8.0.2 release. I think the best option would be exposing all feature bands as `dotnet_8-bin.sdk_x` (so currently `dotnet_8-bin.sdk_1` and `dotnet_8-bin.sdk_2`) and alias `dotnet_8-bin.sdk` to whatever the latest feature band currently is. We can then drop feature bands [as they become EOL](https://learn.microsoft.com/en-us/dotnet/core/porting/versioning-sdk-msbuild-vs#lifecycle) by throwing like we do with <=5.0 sdks right now.

This has the added benefit for people using nixpkgs to manage their development enviroments as .NET SDK feature bands can have major new features and bugs, see https://github.com/dotnet/sdk/issues/39142 which currently breaks most of the style analyzers in the `2xx` feature band while it still works in the `1xx`.

Reply:
This is a pretty good idea.  8.0.3 just came out, so I'll see if i can teach the update script to pull down each SDK for a corresponding runtime.

Reply:
I made a PR with the feature band idea:  #299822